KEYCLOAK-1963 - Support lookup client resource by client-id.

thomasdarimont opened the pull request:
`ClientResources` can now be looked up via the `/admin/realms/{realm}/clients/by-client-id/{clientId}` endpoint.
Previously users had to fetch all clients from the clients recource first and to iteratve over the 
returned client resources in order to find the actual client resource with the matching client-id.

This change spares the client some round-trips.

Note:
If you want to use this PR I'd recommend to just cherry-pick the commit: 9529e39.
Since I work with many different feature branches locally I want to have dedicated versions of the build artifacts hence I adjust the maven project version to $CURRENT_VERSION-JIRA_ISSUE-SNAPSHOT.
in the first branch commit (which then should be ignored for merge).

I set the maven version via:

```
mvn -q versions:set -DgenerateBackupPoms=false -DnewVersion=1.6.0.KEYCLOAK-1963-SNAPSHOT
```

thomasdarimont committed:
KEYCLOAK-1963 - Support lookup client resource by client-id.

Prepare issue branch.

thomasdarimont committed:
KEYCLOAK-1963 - Support lookup client resource by client-id.

`ClientResources` can now be looked up via the ```/admin/realms/{realm}/clients/by-client-id/{clientId}``` endpoint.
Previously users had to fetch all clients from the clients recource first and to iteratve over the returned client
resources to find the actual client resource with the matching client-id.

This change spares the client some round-trips.

stianst commented:
There's two issues with this PR:
* We made the decision to base all admin endpoint on id for a while back
* You've also updated the version in pom files

For now I'm closing the issue. If you want to discuss this further please send a mail to dev mailing list. As a rule it's always good to discuss issues there prior to creating a PR as there may be reasons why we would reject your work.

thomasdarimont commented:
Thanks for the feedback - I removed the version updates (sorry I'm used to this quite alot) and updated my branch. 

I can issue a new PR for this but I'll open a new thread on the mailing list first as you suggested.
